models/quote-request/src/transformers.ts

```diff
@@ -139,7 +133,7 @@ const transformers = {
 
       const cartRef: TReferenceGraphql = Reference.presets.cartReference
         .cartReference()
-        .id(fields.cart?.id)
+        .id(fields.cart?.id || '')
```

Comment:
Ninja move!

Reply:
Something like
```
      const cartRef: TReferenceGraphql | null = fields.cart
        ? Reference.presets.cartReference
            .cartReference()
            .id(fields.cart.id)
            .typeId('cart')
            .buildGraphql()
        : null;
```
could be more in line with the schema, that way we don't have to substitute empty strings and we know `id` is available

Reply:
(full disclosure I did that [here](https://github.com/commercetools/test-data/blob/PANGOLIN-3184-staged-quote-model/models/order/src/transformers.ts#L73-L78))

Reply:
@jaikamat updated